packages/paste-core/components/button/src/ResetButton.tsx

```diff
@@ -29,9 +29,9 @@ const ResetButton = React.forwardRef<HTMLButtonElement, DirectButtonProps>(
       <Box
         ref={ref}
         width={fullWidth ? '100%' : 'auto'}
-        {...safelySpreadBoxProps(props)}
         {...ButtonStyleMapping[buttonState]}
         {...SizeStyles[size]}
+        {...(props as any)}
```

Comment:
question: don't we have to make `DirectButtonProps` extend `BoxProps` in order for reset buttons to accept other props in ts?

Reply:
It's a little more complicated than that. First, the props passed to Button go through `ButtonProps` which we don't want to change. Those props then get passed down to the varianted button component internally which use DirectButtonProps. DirectButtonProps can't extend HTMLButtonElement and BoxProps at the same time to begin with, but even if we managed it fix that somehow, TS will never expect ResetButton to get any of the BoxProps anyways because the parent component has tighter typings.

This would need a larger rethink, but it's fine for now imo.